spyder_kernels/console/shell.py

```diff
@@ -44,7 +47,18 @@ def _showtraceback(self, etype, evalue, stb):
             stb = ['']
         super(SpyderShell, self)._showtraceback(etype, evalue, stb)
 
-    # ---- For Pdb namespace integration
+    def enable_matplotlib(self, gui=None):
+        """Enable matplotlib."""
+        if gui is None or gui.lower() == "auto":
+            gui = automatic_backend()
+        gui, backend = super(SpyderShell, self).enable_matplotlib(gui)
```

Comment:
Perhaps is better to use the methods we have in the kernel to set the Matplotlib backend, which handle errors gracefully? Or do you think that's not necessary in this case?

Reply:
This is the function currently called whenever the matplotlib backend is changed. I reimplement it to notify the backend. The methods we have in the kernel depends on this.